storage/database.go

```diff
@@ -120,9 +141,24 @@ func NewDatabase(shardSet m3db.ShardSet, opts m3db.DatabaseOptions) Database {
 		openCh:       make(chan struct{}, 1),
 		doneCh:       make(chan struct{}, dbOngoingTasks),
 	}
-	database.bsm = newBootstrapManager(database)
-	database.fm = newFlushManager(database)
-	return database
+	d.bsm = newBootstrapManager(d)
+	d.fm = newFlushManager(d)
+
+	var err error
+	if d.commitLog, err = commitlog.NewCommitLog(opts); err != nil {
+		return nil, err
+	}
+
+	switch opts.GetCommitLogStrategy() {
```

Comment:
Definitely in favor of validating as many params as possible in an options `Validate()` method.  This particular instance where the commit log is created and file access to the commit log is check in the constructor however I'd rather not put inside the options.